templates/common/vsphere/files/vsphere-NetworkManager-kni-conf.yaml

```diff
@@ -3,5 +3,9 @@ mode: 0644
 path: "/etc/NetworkManager/conf.d/99-kni.conf"
 contents:
   inline: |
+    {{ if .Infra.Status.PlatformStatus.VSphere -}}
+    {{ if .Infra.Status.PlatformStatus.VSphere.APIServerInternalIP -}}
```

Comment:
hmm there is no way to do AND in templates?

Reply:
Templates does have `and` but when I tested them
```
[root@bootstrap-0 bootstrap-secrets]# crictl logs -f 517
I0205 01:41:15.357524       1 bootstrap.go:40] Version: machine-config-daemon-4.3.0-201910280117-238-g1a379174 (1a379174f8a60cbb57436769fa2b33fed30d0a49)
F0205 01:41:15.412264       1 bootstrap.go:47] error running MCC[BOOTSTRAP]: failed to create MachineConfig for role master: failed to execute template: template: /etc/mcc/templates/common/vs
phere/files/vsphere-NetworkManager-kni-conf.yaml:6:57: executing "/etc/mcc/templates/common/vsphere/files/vsphere-NetworkManager-kni-conf.yaml" at <.Infra.Status.PlatformStatus.VSphere.APISer
verInternalIP>: can't evaluate field APIServerInternalIP in type *v1.VSpherePlatformStatus
[root@bootstrap-0 bootstrap-secrets]# Connection to 139.178.89.200 closed by remote host
```

```
  1 filesystem: "root"
  2 mode: 0644
  3 path: "/etc/NetworkManager/conf.d/99-kni.conf"
  4 contents:
  5   inline: |                                                                                                                                                                                
  6     {{ if and .Infra.Status.PlatformStatus.VSphere .Infra.Status.PlatformStatus.VSphere.APIServerInternalIP -}}
  7     [main]
  8     dhcp=dhclient
  9     {{ end -}}
```
And from the documentation:

> and
> 	Returns the boolean AND of its arguments by returning the
> 	first empty argument or the last argument, that is,
> 	"and x y" behaves as "if x then y else x". **All the
> 	arguments are evaluated.**